lib/e3/ecl/fortio.h

```diff
@@ -0,0 +1,183 @@
+ * the error, same rules as that of eclfio_get.
+ *
+ * This function does not distinguish seek errors for any n not +-1, so to
+ * figure out which record fails, one record at a time must be skipped.
+ */
+int eclfio_skip( FILE*, const char* opts, int n );
+
+/*
+ * Get the next record, and its number of elements.
+ *
+ * The record buffer is generally assumed to be of approperiate size, which can
+ * be queried with eclfio_sizeof.
+ *
+ * On success, the value of recordsize denotes the number of elements read,
+ * whose size is determined by the "cifd" options. It is generally assumed that
+ * recordsize upon calling this function contains the size of the record
+ * buffer, as a failsafe mechanism - if a record is larger than this value, the
+ * read will be aborted and the file position rolled back. To opt out of this
+ * check, add # to opts.
+ *
+ * Both recordsize and record can be NULL, in which case the number of elements
+ * read is not returned, and no data is returned respectively. This allows
+ * precise reporting on how many elements each skipped records contains.
+ *
+ * If the elementsize is larger than 1, and transformation has not been
+ * explicitly disabled, endianness will be converted appropriately.
+ *
+ * It is assumed that all record has an appropriate head and tail. If it is
+ * know that no record has a tail, force this by passing ~ in opts. However, if
+ * it is uncertain if all records has tails, or it's alternating between tail
+ * and no-tail, the $ option tries to recover from missing tails by assuming
+ * the current position is the start of the next record.
+ *
+ * The contents of record* is unspecified in case of read failures, and may not
+ * be relied upon. If the function returns ECL_EINVAL, the output record is
+ * untouched.
+ *
+ * This function returns ECL_OK upon success, ECL_ERR_READ in case of read- or
+ * seek errors, ECL_INVALID_RECORD if either the record tail is broken and
+ * options is set accordingly. The list of error codes is not exhaustive, and
+ * robust code should have fallthrough error handling cases.
+ */
+int eclfio_get( FILE*, const char* opts, int32_t* recordsize, void* record );
+
+/*
+ * Put a record of nmemb elements
+ *
+ * This function will write both head and tail, unless tail writing is
+ * explicitly disabled with ~. If (nmemb * elemsize) overflows int32, the write
+ * is aborted and ECL_EINVAL is returned.
+ *
+ * put largely follows the same rules as get, including those of endianness.
+ * The file pointer is rolled back if any part of the function should fail, as
+ * if the function was never called.
+ *
+ * If a write fails after partial writes, no attempts are made to roll back
+ * written changes.
+ *
+ * Returns ECL_OK on success, or ECL_ERR_WRITE on failure. If ECL_ERR_SEEK is
+ * returned, the integrity of the file stream can not be guaranteed, and its
+ * state is considered unspecified.
+ */
+int eclfio_put( FILE*, const char* opts, int nmemb, const void* );
```

Comment:
I agree that the `nmemb` is the natural parameter here, but just something to be aware of: Strings in eclipse can be stored in two different ways; the most common is 8 character arrays (with space padding):

```
WGNAMES = ["WELL1   ", "WELL_123", "GRPABC  "]
```
When passing this to current `eclfio_put` function the `nmemb` variable must be 24. I.e. there is a difference between `opts="d"` where the internal layer knows the size of the `double`, and the strings where outer layer must convert to a char representation - sizewise - not that I think there is an alternative though.   

In addition to the `'CHAR'` datatype which amounts to fixed 8 character strings there is also support for `Cnnn` strings which are fixed size, space padded, strings of length `nnn`.